### PR TITLE
Prohibit non-ASCII characters in rule metadata.json files

### PR DESCRIPTION
**Pull request by arseniy-sonar**
Triggered by the deployment failure that was caused by an invisible Unicode character in a rule's metadata.json.
This PR implements three conceptual changes:
- make the deployment parse error more informative
- prohibit the use of non-ASCII characters in the metadata.json files
- remove the existing non-ASCII characters from the existing rules

### CODE REVIEW

File: frontend/src/deployment/metadata.ts [hunk 1]
```diff
@@ -117,7 +117,12 @@ function getRuleMetadata(srcDir: string, language?: string) {
     }
     const languageFile = path.join(srcDir, language, 'metadata.json');
     if (fs.existsSync(languageFile)) {
-      return JSON.parse(fs.readFileSync(languageFile, 'utf8'));
+      try {
+        return JSON.parse(fs.readFileSync(languageFile, 'utf8'));
+      } catch (error) {
+        console.error("When parsing the JSON file " + languageFile);
```
Comment (marco-antognini-sonarsource) [inline frontend/src/deployment/metadata.ts:123]:
```suggestion
        console.error('When parsing the JSON file ' + languageFile);
```
That should address the code smell.

But maybe it's also worth printing the exception? I'm not sure how it will be displayed at the moment.

File: rules/S4225/vbnet/metadata.json [hunk 2]
```diff
@@ -1,3 +1,2 @@
-﻿{
-  "title": "Extension methods should not extend \"Object\""
```
Comment (marco-antognini-sonarsource) [inline rules/S4225/vbnet/metadata.json:2]: Is this change intentional? It seems unrelated at first sight.

**Review by marco-antognini-sonarsource — CHANGES_REQUESTED** [review]
It's also nice to make quoting more consistent. 👍

File: rules/S4225/vbnet/metadata.json (hunk 2 above)
Comment (arseniy-sonar) [reply rules/S4225/vbnet/metadata.json:2]: Oups, no. thanks for catching this !

File: frontend/src/deployment/metadata.ts (hunk 1 above)
Comment (arseniy-sonar) [reply frontend/src/deployment/metadata.ts]: The error will be printed together with the stacktrace just below this, because it is rethrown.
This change address the issue that when this exception happens it does not say which file has broken syntax.

**Review by marco-antognini-sonarsource — APPROVED** [review]
LGTM

**Review by marco-antognini-sonarsource — APPROVED** [review]
Thanks for the test, looks good.
